### PR TITLE
Update wasi dependency

### PR DESCRIPTION
**Pull request by newpavlov**

### CODE REVIEW

File: src/wasi.rs [hunk 1]
```diff
@@ -12,11 +12,9 @@ use core::num::NonZeroU32;
...
-    if let Some(code) = NonZeroU32::new(ret as u32) {
-        error!("WASI: random_get failed with return value {}", code);
-        Err(Error::from(code))
-    } else {
-        Ok(()) // Zero means success for WASI
-    }
+    // wasi uses NonZeroU16 for error codes
+    // here compiler is able to eliminate `unwrap`
+    random_get(dest).map_err(|e| {
+        NonZeroU32::new(e.get() as u32).unwrap().into()
```
[the remark this answers is not in the document]
Comment (newpavlov) [reply src/wasi.rs]: >Do we want to keep the error! log for WASI?

Not sure, we don't log all targets, so I guess we can omit it here.

File: src/error.rs [hunk 2]
```diff
@@ -60,22 +60,29 @@ impl Error {
...
-        return None;
-    }
-
-    // Take up to trailing null byte
-    let n = buf.len();
-    let idx = buf.iter().position(|&b| b == 0).unwrap_or(n);
-    core::str::from_utf8(&buf[..idx]).ok()
-}
+cfg_if! {
+    if #[cfg(unix)] {
```
[the remark this answers is not in the document]
Comment (newpavlov) [reply src/error.rs]: `libc` already has [dropped](https://github.com/rust-lang/libc/blob/master/src/lib.rs#L129) support for non-`unix` Redox, so I don't see why should we keep it on our side.

**Comment by josephlr** [comment]
@newpavlov, normally I would merge this, but it seems I am not an authorized user.

**Comment by newpavlov** [comment]
Yeah, I am also not authorized to merge. Probably @dhardy has changes some repository settings.

**Comment by dhardy** [comment]
Aha, didn't plan to revoke any permissions! I just bumped you both to "maintain" status; any good?

**Comment by newpavlov** [comment]
Yeah, it works now. Thanks!

**Comment by mati865** [comment]
Hello, could you soon release new version to unblock https://github.com/rust-lang/rust/pull/63806?

Thanks in advance.

**Comment by newpavlov** [comment]
@mati865 
Done!

**Pull request merged**